src/server/routes/units.js

```diff
@@ -1,258 +1,146 @@
+   async (req, res) => {
+	 const conn = getConnection();
+	 try {
+	   const rows = await Unit.getAll(conn);
+	   res.json(rows.map(item => ({
+		 id: item.id,
+		 name: item.name,
+		 identifier: item.identifier,
+		 unitRepresent: item.unitRepresent,
+		 secInRate: item.secInRate,
+		 typeOfUnit: item.typeOfUnit,
+		 suffix: item.suffix,
+		 displayable: item.displayable,
+		 preferredDisplay: item.preferredDisplay,
+		 note: item.note
+	   })));
+	 } catch (err) {
+	   log.error(`Error fetching units: ${err}`, err);
+	   res.sendStatus(500);
+	 }
+   }
+ );
+ 
 /**
- * Route for POST, edit unit.
+ * Route for editing a unit by ID.
  */
-router.post('/edit', async (req, res) => {
-	const validUnit = {
-		type: 'object',
-		required: ['id', 'identifier'],
-		properties: {
-			id: {
-				type: 'integer'
-			},
-			name: {
-				type: 'string',
-				minLength: 1
-			},
-			identifier: {
-				type: 'string'
-			},
-			unitRepresent: {
-				type: 'string',
-				minLength: 1,
-				enum: Object.values(Unit.unitRepresentType)
-			},
-			secInRate: {
-				type: 'number',
-			},
-			typeOfUnit: {
-				type: 'string',
-				minLength: 1,
-				enum: Object.values(Unit.unitType)
-			},
-			suffix: {
-				type: 'string'
-			},
-			displayable: {
-				type: 'string',
-				minLength: 1,
-				enum: Object.values(Unit.displayableType)
-			},
-			preferredDisplay: {
-				type: 'bool'
-			},
-			note: {
-				type: 'string'
-			},
-			minVal: {
-				type: 'number',
-			},
-			maxVal: {
-				type: 'number',
-			},
-			disableChecks: {
-				type: 'string',
-				minLength: 1,
-				enum: Object.values(Unit.disableChecksType)
-			}
-		}
-	};
-	const validatorResult = validate(req.body, validUnit);
-	if (!validatorResult.valid) {
-		log.warn(`Got request to edit units with invalid unit data, errors:${validatorResult.errors}`);
-        failure(res, 400, `Got request to edit units with invalid unit data, errors:${validatorResult.errors}`);
-	} else {
-		const conn = getConnection();
-		try {
-			const unit = await Unit.getById(req.body.id, conn);
-			if (unit.suffix !== req.body.suffix) {
-				// Suffix changes so some conversions and units need to be removed.
-				await removeAdditionalConversionsAndUnits(unit, conn);
-			}
-			unit.name = req.body.name;
-			unit.displayable = req.body.displayable;
-			unit.identifier = req.body.identifier;
-			unit.unitRepresent = req.body.unitRepresent;
-			unit.typeOfUnit = req.body.typeOfUnit;
-			unit.preferredDisplay = req.body.preferredDisplay;
-			unit.secInRate = req.body.secInRate;
-			unit.suffix = req.body.suffix;
-			unit.note = req.body.note;
-			unit.minVal = req.body.minVal;
-			unit.maxVal = req.body.maxVal;
-			unit.disableChecks = req.body.disableChecks;
-			await unit.update(conn);
-		} catch (err) {
-			log.error('Failed to edit unit', err);
-            failure(res, 500, 'Unable to edit unit ' + err.toString());
-		}
-		success(res, `Successfully edited unit`);
-	}
-});
-
+ router.post(
```

Comment:
There are a lot of changes in the remainder of this file including removal of the request parameter validation, some comments (some changes might be okay but others are not) and response messages. I'm unclear why they were made. Could you either explain of put them back. I'll need to review this file again once this is settled.